packages/search-common/src/types.ts

```diff
@@ -68,6 +68,12 @@ export interface IndexableDocument {
   };
 }
 
+/**
+ * Properties of {@link IndexableDocument}s that are relevant to frontend
+ * consumers.
+ */
+export type SearchDocument = Omit<IndexableDocument, 'authorization'>;
```

Comment:
Mmm - for what it's worth, I'd probably assert that even if the IndexableDocument type were generic, the authorization property should still be consistently included on it. 

That said, even if it were generic, would that save casting on the frontend? Wouldn't you still by default end up getting something like `IndexableDocument<unknown>` from the query, and have to cast it to `ElasticSearchIndexableDocument` anyway?

Reply:
> Mmm - for what it's worth, I'd probably assert that even if the IndexableDocument type were generic, the authorization property should still be consistently included on it.
Got it, make sense to have the authorization in the default properties.
> That said, even if it were generic, would that save casting on the frontend? Wouldn't you still by default end up getting something like IndexableDocument<unknown> from the query, and have to cast it to ElasticSearchIndexableDocument anyway?

Maybe the [Search API](https://github.com/backstage/backstage/blob/db2666b980853c281b8fe77905d7639c5d255f13/plugins/search/src/apis.ts#L31) implementation can have a generic type for its document set, then the query method will return the correct type of results, but I'm not sure about the [SearchResult](https://github.com/backstage/backstage/blob/4045eaab952fec349e443990330bd1ac5b2c9c18/plugins/search/src/components/SearchResult/SearchResult.tsx#L19) component we have today, it depends on a non-generic type forcing us to cast always. Would be a nice developer experience if this type could be inferred, but it may not be possible?

Reply:
Yeah - I think that's the key. It might be that generics are a good way to type these things, but there's other options too, including a type which extends the existing IndexableDocument interface (kinda like the [TechDocsDocument](https://github.com/backstage/backstage/blob/master/packages/techdocs-common/src/techdocsTypes.ts#L19) type). I think this could be an interesting area to explore - it seems to me that doing some kind of spike to figure out what criteria should be used to refine the types and which apis/components should do the refinement might be a good next step.

Reply:
Agree, thanks for explaining, looking forward to following these changes!